### PR TITLE
Allow application supporters to use route endpoints

### LINKED ISSUE #2229: Allow space application supporter to access specific route endpoints.
## Issue

Allow space application supporter to access specific **route** endpoints. 

## Context

We are introducing a [new role](https://github.com/cloudfoundry/cfar-proposals/issues/22) and we want to make sure it has the right access.

## Expected result
A space application support should be able access the following endpoints:

POST /v3/routes
GET /v3/routes
DELETE /v3/routes/:guid
GET /v3/domains/:guid/route_reservations
GET /v3/routes/:guid/destinations
GET /v3/apps/:guid/routes
POST /v3/routes/:guid/destinations
PATCH /v3/routes/:guid/destinations
DELETE /v3/routes/:guid/destinations/:destination_guid
DELETE /v3/spaces/:guid/routes?unmapped=true

## Acceptance
A space application supporter would see the same info as a space developer assigned to the same space for these and only these routes endpoints.

## Documentation
When I browse to any of these endpoints on [v3 docs](https://v3-apidocs.cloudfoundry.org/version/3.99.0/index.html) I can see the Space Application Supporter role in the list of permitted roles with an indication that this role is not fully implemented and the permissions will be changing.

### PR DESCRIPTION
**Pull request by andy-paine**
The readable_org_guids_for_domains function takes a flag for now about
whether to include application supporters. All usages of this function
will eventually set this flag to true at which point it can be removed.

As some controllers start using the untrusted_* functions, the
user_helpers spec file needs to return the correct value for whichever
of these functions has been called. The tests that use this helper do
not need to know which of the two functions was called, only that the
permission will or will not be granted.

Added a has_application_supporter? public function to the space object
so that we can use untrusted_can_read_route? as the can_read_route?
function is used by v2 endpoints which we don't want to grant access to
for application supporters

Co-authored-by: Philipp Thun <philipp.thun@sap.com>

Closes #2229 

* [x] I have reviewed the [contributing guide](https://github.com/cloudfoundry/cloud_controller_ng/blob/main/CONTRIBUTING.md)

* [x] I have viewed, signed, and submitted the Contributor License Agreement

* [x] I have made this pull request to the `main` branch

* [x] I have run all the unit tests using `bundle exec rake`

* [ ] I have run [CF Acceptance Tests](https://github.com/cloudfoundry/cloud_controller_ng#cf-acceptance-tests-cats)

### CODE REVIEW

**Review by monamohebbi — APPROVED** [review]
LGTM

We have deployed changes to a test environment and ran through the basic commands with a [script](https://github.com/monamohebbi/RBAC_acceptance/blob/main/README.md
) that can be modified for any local environment to test changes as well. Now just waiting on the official maintainers to review and merge this as well.

**Review by sweinstein22 — APPROVED** [review]
Looks good, thanks for the PR!

**Pull request merged**
